admin/includes/init_includes/init_check_parms.php

```diff
@@ -0,0 +1,15 @@
+<?php
+/**
+ * @package admin
+ * @copyright Copyright 2003-2015 Zen Cart Development Team
+ * @copyright Portions Copyright 2003 osCommerce
+ * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
+ */
+if (!defined('IS_ADMIN_FLAG')) {
+  die('Illegal Access');
+}
+if (isset($_GET['product_type'])) { 
+  if (zen_get_handler_from_type((int)$_GET['product_type']) == -1) {
+      unset($_GET['product_type']); 
+  }
```

Comment:
To avoid the arrow anti-pattern, let's combine the `if isset` with the `if zen_get_handler_from_type` using `&&`